### PR TITLE
Docker: Move apt-get commands to earlier and add apt-get update 

### PR DESCRIPTION
**Pull request by redfuture**
before the install.

apt-get commands part is more stable than the installer part, which including build
steps,this may could fasten the docker image build process when there are only
installers changed.

### CODE REVIEW

**Comment by xiaoxq** [comment]
Good point!

Actually installers/pre_install.sh did this part. Could you move the content out and remove the pre_install.sh.
Or the other way, move the commands into pre_install.sh and run it first:
```
COPY installers/pre_install.sh /tmp/installers/pre_install.sh
RUN bash /tmp/installers/pre_install.sh

COPY installers /tmp/installers
...
```

**Comment by redfuture** [comment]
Thanks for your review, I didn't notice the pre_install.sh , I will make  another patch to move out the pre_install.sh content out and merge with the apt-get commands, this looks like more readable.

**Comment by redfuture** [comment]
I pushed again to the same branch , please review. thanks.

File: docker/build/dev.x86_64.dockerfile [hunk 1]
```diff
@@ -2,37 +2,45 @@ FROM nvidia/cuda:8.0-cudnn7-devel-ubuntu14.04
...
-   realpath \
-   shellcheck \
-   vim \
-   v4l-utils \
-   nfs-common \
-   zip && \
-   apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN apt-get update -y && \
+    apt-get install -y \
+    apt-transport-https \
```
Comment (xiaoxq) [inline docker/build/dev.x86_64.dockerfile]: Please make this list in sorted order. So that it's easier for future lookup and modification.

Comment (redfuture) [reply docker/build/dev.x86_64.dockerfile]: thanks for your quick review, I sorted the list and pushed another commit on the same branch. Please review again.

**Comment by xiaoxq** [comment]
Excellent, thanks!

**Pull request merged**